src/Chunk.ts

```diff
@@ -110,11 +110,12 @@ export default class Chunk {
 	// an input entry point module
 	entryModule: Module;
 	isEntryModuleFacade: boolean;
+	isManualChunk: boolean;
 
-	renderedHash: string;
-	renderedSources: MagicString[];
-	renderedSource: MagicStringBundle;
-	renderedDeclarations: { dependencies: ChunkDependencies; exports: ChunkExports };
+	private renderedHash: string;
+	private renderedSources: MagicString[];
```

Comment:
I like the idea of separation here bringing more flexibility here, but one thing that worries me is attaching too much rendered state into Module. Already we attach rendered state into the AST itself which I don't really like doing (would be nicer to treat this as a more separated layer less dependent on execution order, it also binds us to synchronous rendering).

So I agree with the direction, but I'm not sure it would be the right implementation. Perhaps a `Map` would be best, if performance can work out for that.